### ISSUE TITLE
Questions about training and testing rnn-classification model

### ISSUE THREAD

**Issue by alisonswu** [spro/practical-pytorch#11]
Thank you for creating these tutorials! I am new to both RNN and pytorch. I found your tutorials to be incredibly helpful for getting started with the topic.

I have a few high level questions about training and testing the classification model in the char-rnn-classification tutorial. 

1. The data size is unbalanced between different language categories.  The smallest category is Vietnamese with 73 names, whereas the largest category is Russian with  9408 names. Your model handles the unbalanced data by sampling names with equal probability from each language category to train the RNN.  My question is:
- Are there other practical solution to handle unbalanced classes in training a RNN classifier? 
- Theoretically, would weighting the loss of each line tensor inversely to their category size help?

2. The entire data has 20,074 names across all language categories. The RNN is trained with 100,000 names sampled with replacement from the data. Some names are used to train the RNN multiple times. My question is:
- Is this done in propose the resample from the underrepresented language categories?
- Would this cause model over-fitting?

Thanks!

**Comment by spro** [comment]
1. This technique (over/under sampling) is my preference because of the simplicity. Class weights (which you can [specify with NLLLoss](http://pytorch.org/docs/nn.html#nllloss)) should be another way to deal with the unbalance. Certain problems might even make use of synthetic data to fill in the gaps (especially image classification).

2. Yes and maybe. It gives the model a chance to see all examples possibly several times, which helps it better fit the data. However the model is small enough (2 layers with 128 hidden units) that it will probably not *over*fit.

**Comment by alisonswu** [comment]
Thank you so much for your response!
